### PR TITLE
rbd-mirror: non-primary image is recording journal events during image sync

### PR DESCRIPTION
**Pull request by dillaman**

### CODE REVIEW

**Comment by trociny** [comment]
@dillaman Could you please rebase?

File: src/test/librbd/test_mock_AioImageRequest.cc [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+  mock_image_ctx.journal = &mock_journal;
+
+  InSequence seq;
+  expect_is_journal_appending(mock_journal, false);
+  expect_flush(mock_image_ctx, 0);
+
+  C_SaferCond aio_comp_ctx;
+  AioCompletion *aio_comp = AioCompletion::create(&aio_comp_ctx);
+  aio_comp->init_time(ictx, AIO_TYPE_FLUSH);
+  aio_comp->start_op();
```
Comment (trociny) [inline src/test/librbd/test_mock_AioImageRequest.cc]: @dillaman AioCompletion::create_and_start?

**Comment by trociny** [comment]
lgtm

**Comment by dillaman** [comment]
@trociny pushed test update

**Pull request merged**